LinqToRest/ResourceRetriever.cs

```diff
@@ -106,9 +108,42 @@ private static Uri GetResourceUri(JToken token, Uri root)
         private object GetDeserializedParameter(ParameterInfo parameter, JToken token, Uri uri)
         {
             var type = parameter.ParameterType;
+
             return type.IsQueryable()
                 ? _queryableFactory(type.GetInnerType(), uri)
-                : GetField(type, token, parameter.Name);
+                : type.IsEnum
+                    ? GetEnumFromType(type, token, parameter.Name)
+                    : GetField(type, token, parameter.Name);
+        }
+
+        private static object GetEnumFromType(Type type, JToken token, string name)
+        {
+            var getEnum = typeof(ResourceRetriever).GetMethod(nameof(GetEnum))?.MakeGenericMethod(type);
+
+            try
+            {
+                return getEnum is null
+                    ? throw new InvalidOperationException()
+                    : getEnum.Invoke(null, new object[] {token, name});
+            }
+            catch (System.Reflection.TargetInvocationException e)
+            {
+                if (e.InnerException != null)
+                {
+                    throw e.InnerException;
+                }
+                throw;
+            }
+        }
+
+        public static T GetEnum<T>(JToken token, string name) where T : struct
+        {
+            var candidate = GetField(typeof(string), token, name) as string;
+            var parsed = candidate.TryParseEnum<T>();
+
+            return parsed.Match(false, e => true)
+                ? parsed.Match(default(T), e => e)

```

Comment:
Why do you `Match` twice on the same value? Wouldn't it make more sense to write something like 
```csharp
parsed.Match(
    none: throw new InvalidEnumArgumentException(name),
    some: @enum => @enum
)
```
?

Reply:
Writing it like that would be perfect but ist isn't possible with funcky atm.
But I improved the naming https://github.com/messerli-informatik-ag/linq-to-rest/pull/15/commits/5757c246c8cdb57e3e94741879f2093b9801d988

Reply:
Why not? Is it a type mismatch? How about returning a function then? 
```csharp
none: () => throw new InvalidEnumArgumentException(name),
some: @enum => () => @enum
```
If that doesn't work, is there any other way to write this? I really dislike the chained `.Match` statements on the same value

Reply:
I agree. Thomas is currently adding overloads to the match function to allow throwing exceptions in match statements